pkg/compactor/compactor.go

```diff
@@ -99,32 +98,42 @@ func (cp *Compactor) Compact(opts *brtypes.CompactOptions) (*brtypes.Snapshot, e
 	// Then compact ETCD
 
 	// Build Client
-	cfg := clientv3.Config{MaxCallSendMsgSize: cmpctOptions.Config.MaxCallSendMsgSize, Endpoints: ep}
-	client, err := clientv3.New(cfg)
+	clientFactory := etcdutil.NewClientFactory(opts.RestoreOptions.NewClientFactory, brtypes.EtcdConnectionConfig{
+		MaxCallSendMsgSize: cmpctOptions.Config.MaxCallSendMsgSize,
+		Endpoints:          ep,
+		InsecureTransport:  true,
+	})
+	clientKV, err := clientFactory.NewKV()
 	if err != nil {
-		return nil, fmt.Errorf("failed to build client")
+		return nil, fmt.Errorf("failed to build etcd KV client")
 	}
-	defer client.Close()
+	defer clientKV.Close()
+
+	clientMaintenance, err := clientFactory.NewMaintenance()
+	if err != nil {
+		return nil, fmt.Errorf("failed to build etcd maintenance client")
+	}
+	defer clientMaintenance.Close()
```

Comment:
This is used only for defragmentation. Can you please move it inside the `if opts.NeedDefragmentation {}` block?

Reply:
yes, make sense I will do it

Reply:
refer https://github.com/gardener/etcd-backup-restore/pull/381#discussion_r712148016